### PR TITLE
Default to Gutenberg upon opening a block based post

### PR DESCRIPTION
**Pull request by marecar3**
This is the Android part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1186

It flips the "default for new posts" switch to true the first time the block editor is launched if the switch is off.

### CODE REVIEW

**Comment by daniloercoli** [comment]
Should we default to Gutenberg upon opening a block based post only once?

With this patch the app does inform me of the changes made to default editor setting, and open the GB based post in GB mobile. That's fine. The block editor is now the default for new posts.
Later, if I go to app settings and turn the block editor OFF for new post, the app should not show the dialog and make the switch once again when I open a block based post, no? it should just open it in GB-mobile, without making any changes to the default editor setting.

**Comment by iamthomasbishop** [comment]
> Later, if I go to app settings and turn the block editor OFF for new post, the app should not show the dialog and make the switch once again when I open a block based post, no? it should just open it in GB-mobile, without making any changes to the default editor setting.

I think that's correct, when an existing GB-based post is opened we don't need to show the dialog.

**Comment by marecar3** [comment]
Thanks @daniloercoli for bringing this one and @iamthomasbishop for confirming this one 👍 
Will fix it.

**Comment by marecar3** [comment]
I have another thought. Should we then present previous popup @iamthomasbishop where it says that we have enabled Gutenberg for current Post only?

**Comment by daniloercoli** [comment]
Seems that the discussion about the new message is ongoing on the main issue https://github.com/wordpress-mobile/gutenberg-mobile/issues/1186

**Comment by marecar3** [comment]
> Should we default to Gutenberg upon opening a block based post only once?
> 
> With this patch the app does inform me of the changes made to default editor setting, and open the GB based post in GB mobile. That's fine. The block editor is now the default for new posts.
> Later, if I go to app settings and turn the block editor OFF for new post, the app should not show the dialog and make the switch once again when I open a block based post, no? it should just open it in GB-mobile, without making any changes to the default editor setting.
> 
> cc @hypest @iamthomasbishop

Hey, @daniloercoli I have fixed this one. You can do another iteration of testing :)

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java [hunk 1]
```diff
@@ -646,17 +646,21 @@ public static boolean isGutenbergEditorEnabled() {
         return true;
     }
 
-    public static void enableGutenbergEditor(boolean enabled) {
-        setBoolean(DeletablePrefKey.GUTENBERG_EDITOR_ENABLED, enabled);
+    public static boolean shouldAutoEnableGutenbergForTheNewPosts() {
+        return getBoolean(DeletablePrefKey.SHOULD_AUTO_ENABLE_GUTENBERG_FOR_THE_NEW_POSTS, true);
+    }
+
+    public static void autoEnableGutenbergForThewNewPosts(boolean enable) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: It seems there is a typo here.

**Comment by daniloercoli** [comment]
@etoledom has made a very small change to the text  https://github.com/wordpress-mobile/gutenberg-mobile/issues/1186#issuecomment-508377224 

I think we should use the same wording.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java (hunk 1 above)
Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: What's about using the set/get convention for these new 2 methods? 
`setAutoEnableGutenbergForNewPosts` and `getAutoEnableGutenbergForNewPosts`

Comment (marecar3) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: Thanks @daniloercoli! It's fixed now.

**Comment by marecar3** [comment]
Hey @daniloercoli, I have updated strings based on the latest proposal: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1186#issuecomment-508552891

**Comment by daniloercoli** [comment]
There are conflicts in the Strings file. Feel free to merge this PR once those are fixed.

**Comment by marecar3** [comment]
Hey @daniloercoli, I have resolved conflict. 
Please, can you give me approve so that I can merge this one? Thanks!

**Comment by marecar3** [comment]
Thanks,@daniloercoli!

**Pull request merged**